components/brave_wallet/browser/polkadot/polkadot_tx_manager.cc

```diff
@@ -58,6 +66,79 @@ void PolkadotTxManager::ApproveTransaction(
                           "Not implemented");
 }
 
+namespace {
+
+// TODO(https://github.com/brave/brave-browser/issues/51544): Eventually migrate
+// off of `const std::string&`.
+std::optional<std::array<uint8_t, kPolkadotSubstrateAccountIdSize>>
+ParsePolkadotAccount(const std::string& input) {
+  auto ss58_address = Ss58Address::Decode(input);
+  if (ss58_address) {
+    return ss58_address->public_key;
+  }
+
+  std::array<uint8_t, kPolkadotSubstrateAccountIdSize> pubkey = {};
+  if (base::HexStringToSpan(input, pubkey) ||
+      PrefixedHexStringToFixed(input, pubkey)) {
+    return pubkey;
+  }
+
+  return std::nullopt;
```

Comment:
This should be in some polkadot utility file and be covered with tests.
It's an important and sensitive part of tx workflow as we are parsing user's input and determining where funds will go in blockchain.